eureka_ml_insights/models/models.py

```diff
@@ -366,6 +368,8 @@ def get_response(self, request):
         openai_response = completion.model_dump()
         self.model_output = openai_response["choices"][0]["message"]["content"]
         self.response_time = end_time - start_time
+        if "usage" in openai_response:
```

Comment:
So the function is called get_response but its returning a dictionary with the usage and setting self.response with the response. Its either a misleading name or not sure I follow the convention of some things are set in self object and others are returned.

Reply:
What I meant was it gets the response (and meta data) from the model. model_output and and is_valid are required for all models so they are class variables. Other things are optional, so they are returned just to be passed up to the caller.

Reply:
I see maybe the name `set_response_return_metadata` would be more relevant but its a nit. I would just move away from stateful model impl towards stateless but we can discuss offline. Also, why not store the whole response? if API changes and they start adding more useful metadata this will automatically be handled instead of being forced to update this code to cherry-pick the right values. That way the caller can decide what to do with this whole response instead of just model_output and a select few.

Reply:
Let's discuss offline. I'm concerned about cases where the user uses several inference components in sequence, and needs to rename the result columns of one inf comp (if they want to keep tham and not have them overwritten by the next one). If there is an unknown/arbitrary number of result columns that step becomes harder.